mmengine/dataset/base_dataset.py

```diff
@@ -222,10 +221,11 @@ def __init__(self,
                  test_mode: bool = False,
                  lazy_init: bool = False,
                  max_refetch: int = 1000):
-
-        self.data_root = data_root
-        self.data_prefix = copy.copy(data_prefix)
         self.ann_file = ann_file
+        self._metainfo = self._load_metainfo(copy.deepcopy(metainfo))
+        self.data_root = data_root
+        data_prefix = copy.copy(data_prefix)
+        self.data_prefix = data_prefix
```

Comment:
Why do we need to split `self.data_prefix = copy.copy(data_prefix)` to two lines?